.github/workflows/presubmit.yaml

```diff
@@ -101,6 +101,29 @@ jobs:
       with:
         name: debs
         path: debs.tar
+  e2e-tests-orchestration-build-image:
+    runs-on: ubuntu-24.04
+    if: ${{ always() && needs.build-debian-package.result == 'success' }}
```

Comment:
maybe a dumb question: why is `always()` needed here? In fact, why is the whole conditional needed? IIUC `always()` says that the job should always run, but the other requirement says only if the dep succeeded, which means runs only when the dep succeeds, but that's the default behavior when having a `needs:` clause. Am I missing something?